dali/sasha/saxref.cpp

```diff
@@ -911,17 +951,43 @@ class CNewXRefManager: public CNewXRefManagerBase
                 fname.toLowerCase();
             addPathSepChar(path).append(fname);
             if (iter->isDir())  {
-                if (!dirFiltered(path.str())) {
-                    dirs.append(fname.str());
+                // Check if subdirectory is a dirPerPart or stripe directory
+                const char *dir = fname.str();
+                bool isStriped = dir[0] == 'd';
+                if (isStriped)
+                    dir++;
+                bool isSpecialDir = true;
+                while (isSpecialDir && *dir) {
+                    if (!isdigit(*(dir++)))
+                        isSpecialDir = false;
+                }
+                // If numDevices>0, then pdir == root, and all directories should be stripe directories
+                if (numDevices>0 && (!isSpecialDir || !isStriped))
```

Comment:
is pdir == root condition missing?

See other comment re. checking numDevices once, shouldn't redo per recursed directory.
To clarify distinction between the plane being striped and "isStriped" , can you rename "isDirStriped" here.

Aside from pdir==root, I think this condition is a bit wrong.
what if !isPlaneStriped && isDirStriped .. that should be an error too.

I think this expression needs to be something like:

if ((pdir==root) && (isPlaneStriped != isDirStriped)) // mismatch